tests/nodeenv_test.py

```diff
@@ -130,3 +130,30 @@ def rewind(_):
                 nodeenv.main()
                 mock_urlopen.assert_called_with(url)
                 mock_logger.assert_called()
+
+
+@pytest.mark.usefixtures('mock_index_json')
+def test_get_latest_node_version():
+    assert nodeenv.get_last_stable_node_version() == '13.5.0'
+
+
+@pytest.mark.usefixtures('mock_index_json')
+def test_get_lts_node_version():
+    assert nodeenv.get_last_lts_node_version() == '12.14.0'
+
+
+@pytest.mark.parametrize('version,expect', [
+    ('12.8.0', '12.8.0'),
+    ('latest', '13.5.0'),
+    ('lts', '12.14.0'),
+])
+@pytest.mark.usefixtures('mock_index_json')
+def test_specify_node_version(version, expect):
+    test_argv = [__file__, '-n', version, HERE]
+    with mock.patch.object(sys, 'argv', test_argv), \
+         mock.patch.object(nodeenv, 'get_env_dir'), \
+         mock.patch.object(nodeenv, 'create_environment') as mock_create_env:
+        nodeenv.main()
+        opts = mock_create_env.call_args.args[-1]
+
+        assert opts.node == expect
```

Comment:
I don't think this is a useful test -- you're just reimplementing the implementation in the test with mocks

Reply:
I suppose that this is really testing the if conditions linked below, i.e. the correct version lookup method is being used depending on the input. If any more special version names are added in the future this would be an easily updated test to ensure that the version resolution still works. I think there's some use there but I don't mind deleting it if there's strong pushback on it. https://github.com/ekalinin/nodeenv/blob/cc32402583d7f9314d2052a0a253ef6b7c96edaf/nodeenv.py#L1063-L1066

Reply:
the problem with these style of tests is if you change the implementation you have to change the test in the same way -- it's not really testing anything at that point other than "can I copy paste the implementation twice"